beets/ui/commands.py

```diff
@@ -684,10 +704,18 @@ def choose_match(self, task):
         # Loop until we have a choice.
         candidates, rec = task.candidates, task.rec
         while True:
+            # Gather extra choices from plugins.
+            # TODO: check for conflicts in choices (duplicated letters, etc)
+            # TODO: provide a way to override summary_judment and/or non-timid
+            # mode? Might be useful for some plugins
+            extra_choices = list(chain(*plugins.send('before_choose_candidate',
```

Comment:
Maybe a helper function like `get_plugin_options` to wrap up this list-flattening and such would aid readability?